src/components/ShoppingCart.js

```diff
@@ -2,14 +2,17 @@ import React from "react";
 
 const ShoppingCart = ({ cart, setCart, updateCart, isOpen, setIsOpen }) => {
   function handleDecrementBtn(p) {
-    const newcart = cart.filter((item) => {
-      if (item.id === p.id && item.count > 0) {
-        item.count = item.count - 1;
-      }
-      return item;
-    });
-
-    setCart(newcart);
+    if (p.count > 1) {
+      const newcart = cart.filter((item) => {
```

Comment:
This works, but there is no reason to use `filter` here. Using `map` is more appropriate here because you're not 'filtering' anything. You're looking through an array and changing one of the items. 

`filter` is better used if you're searching for something, or trying to make the array smaller.